### PR TITLE
Upgrade RocksDB to 6.4.x

### PR DESCRIPTION
**Pull request by yiwu-arbug**
Signed-off-by: Yi Wu <yiwu@pingcap.com>

###  What have you changed?
Upgrade RocksDB to 6.4.x.

###  What is the type of the changes?
Engineering

###  How is the PR tested?
CI

###  Does this PR affect documentation (docs) or should it be mentioned in the release notes?
No

###  Does this PR affect `tidb-ansible`?
No

###  Refer to a related PR or issue link (optional)
https://github.com/pingcap/titan/pull/61
https://github.com/pingcap/rust-rocksdb/pull/336

### CODE REVIEW

**Comment by zhangjinpeng1987** [comment]
So many changes in Cargo.lock?

**Comment by zhangjinpeng1987** [comment]
point select has regression, interesting.

**Comment by aknuds1** [comment]
AFAICT, the upgrade_master branch is gone from your rust-rocksdb fork?

File: src/raftstore/store/snap.rs [hunk 1]
```diff
@@ -2398,7 +2398,7 @@ pub mod tests {
             .unwrap();
 
             // TODO: this size may change in different RocksDB version.
-            let snap_size = 1438;
+            let snap_size = 1658;
```
Comment (Connor1996) [inline src/raftstore/store/snap.rs]: what if the snap_size is not changed when upgrading Rocksdb version?

**Comment by yiwu-arbug** [comment]
`raftstore::test_region_heartbeat::test_region_heartbeat_timestamp` fail in CI but passes locally. Is it flaky?

**Comment by siddontang** [comment]
seem the QPS of `oltp_point_select` is still reduced, interesting. 

have we located the cause?

**Comment by yiwu-arbug** [comment]
@siddontang not yet. benchbot result on #5490 show there are ~2% point-select regression from 6.2 to 6.3, and ~3% more regression from 6.3 to 6.4. I was wanting to rerun point-select on a larger data set (benchbot data set size is ~10G), but haven't got a chance to do so yet. Since the regression is not too big, how about we land this patch to fix clang11 compile issue, and investigate the regression later?

cc @Connor1996 @zhangjinpeng1987 @Little-Wallace

**Comment by siddontang** [comment]
I think it is ok.

PTAL @zhouqiang-cl

**Pull request merged**